components/webcompat_reporter/browser/webcompat_reporter_service.cc

```diff
@@ -0,0 +1,193 @@
+    return;
+  }
+  if (!service_delegate) {
+    return;
+  }
+
+  auto component_infos = service_delegate->GetComponentInfos();
+  if (!component_infos) {
+    return;
+  }
+
+  auto components_list = base::Value::List();
+  for (const auto& component : component_infos.value()) {
+    base::Value::Dict component_dict;
+    component_dict.Set(kComponentItemName, component.name);
+    component_dict.Set(kComponentItemId, component.id);
+    component_dict.Set(kComponentItemVersion, component.version);
+    components_list.Append(std::move(component_dict));
+  }
+  report.ad_block_components = base::Value(std::move(components_list));
+}
+
+void FillReportByReportInfo(
+    webcompat_reporter::Report& report,
+    webcompat_reporter::mojom::ReportInfoPtr report_info) {
+  if (!report_info) {
+    return;
+  }
+  report.channel = report_info->channel;
+  report.brave_version = report_info->brave_version;
+
+  if (report_info && report_info->report_url &&
+      !report_info->report_url->empty()) {
+    report.report_url = GURL(report_info->report_url.value());
+  } else {
+    report.report_url = std::nullopt;
+  }
+
+  if (report_info->ad_block_components_version) {
+    ConvertCompsToValue(report.ad_block_components,
+                        report_info->ad_block_components_version.value());
+  }
+
+  if (report_info->ad_block_list_names &&
+      !report_info->ad_block_list_names->empty()) {
+    report.ad_block_list_names = report_info->ad_block_list_names;
+  }
+
+  report.shields_enabled = report_info->shields_enabled;
+  report.ad_block_setting = report_info->ad_block_setting;
+  report.fp_block_setting = report_info->fp_block_setting;
+  report.languages = report_info->languages;
+  report.language_farbling = report_info->language_farbling;
+  report.brave_vpn_connected = report_info->brave_vpn_connected;
+  report.details = report_info->details;
+  report.contact = report_info->contact;
+  report.screenshot_png = report_info->screenshot_png;
+}
+
+void FillReportValues(
+    webcompat_reporter::Report& report,
+    webcompat_reporter::WebcompatReporterService::WebCompatServiceDelegate*
+        service_delegate) {
+  if (!report.channel && service_delegate) {
+    report.channel = service_delegate->GetChannelName();
+  }
+
+  if (!report.brave_version) {
+    report.brave_version =
+        version_info::GetBraveVersionWithoutChromiumMajorVersion();
+  }
+  FillReportWithComponetsInfo(report, service_delegate);
+  FillReportWithAdblockListNames(report, service_delegate);
+}
+}  // namespace
+namespace webcompat_reporter {
+
+WebcompatReporterService::WebcompatReporterService() = default;
+
+WebcompatReporterService::WebcompatReporterService(
+    std::unique_ptr<WebCompatServiceDelegate> service_delegate,
+    scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory)
+    : service_delegate_(std::move(service_delegate)),
+      report_uploader_(
+          std::make_unique<WebcompatReportUploader>(url_loader_factory)) {}
```

Comment:
construct `WebcompatReportUploader` in the factory by passing `network::SharedURLLoaderFactory` there. You don't need `url_loader_factory` here.